### PR TITLE
SARIMAX -> new arch + fixes 

### LINKED ISSUE #3: SARIMAX -> new arch

### PR DESCRIPTION
**Pull request by alex-hse-repository**
Closes #3

### CODE REVIEW

File: etna/models/sarimax.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+            )
+
+        self._check_df(df)
+
+        targets = df["target"]
+        targets.index = df["timestamp"]
+
+        exog_train = self._select_regressors(df)
+        regressor_columns = None
+        if not isinstance(exog_train, type(None)):
```
Comment (julia-shenshina) [inline etna/models/sarimax.py:185]:
```suggestion
        if exog_train is not None:
```

File: etna/models/sarimax.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+            regressor_columns = [col for col in df.columns if col.startswith("regressor")]
```
Comment (julia-shenshina) [inline etna/models/sarimax.py:186]: isn't it just a 
```suggestion
            regressor_columns = exog_train.columns
```
?

File: etna/models/sarimax.py [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+                "Try to encode this columns manually."
+            )
+
+        exog_future = self._select_regressors(df)
+        y_pred = self._result.predict(
+            start=df["timestamp"].min(), end=df["timestamp"].max(), dynamic=True, exog=exog_future
+        )
+        return y_pred.reset_index(drop=True, inplace=False)
+
+    def _check_df(self, df, horizon=None):
```
Comment (julia-shenshina) [inline etna/models/sarimax.py:252]: typings

File: etna/models/sarimax.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/models/sarimax.py:185]: Если exon_train валидный вернется, то этот if упадет

File: etna/models/sarimax.py (hunk 2 above)
Comment (alex-hse-repository) [reply etna/models/sarimax.py]: Сори, я просто на if отвлекся и забыл поправить)

File: tests/test_models/test_sarimax_model.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        train_start=None, train_end="2020-01-31", test_start="2020-02-01", test_end="2020-02-24"
+    )
+    prep = TheilSenTrendTransform(in_column="target")
+    train.fit_transform([prep])
+    model = SARIMAXModel()
+    model.fit(train)
+    future_ts = train.make_future(future_steps=horizon)
+    reg_result = model.forecast(future_ts)
+
+    van_acc = mean_absolute_error(test[:, :, "target"], vanilla_result[:, :, "target"], multioutput="raw_values")
```
Comment (julia-shenshina) [inline tests/test_models/test_sarimax_model.py:65]: in `etna.metrics` we already have MAE that works with ETNA data format

File: etna/models/sarimax.py (hunk 1 above)
Comment (julia-shenshina) [reply etna/models/sarimax.py:185]: ...in both cases? :)

Comment (alex-hse-repository) [reply etna/models/sarimax.py:185]: No, my version handle this correctly

File: etna/models/sarimax.py [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+
+        self._check_df(df)
+
+        targets = df["target"]
+        targets.index = df["timestamp"]
+
+        exog_train = self._select_regressors(df)
+        regressor_columns = None
+        if not isinstance(exog_train, type(None)):
+            regressor_columns = exog_train.columns
```
Comment (alex-hse-repository) [inline etna/models/sarimax.py:186]:
```suggestion
            regressor_columns = exog_train.columns.values
```

File: etna/models/sarimax.py (hunk 1 above)
Comment (julia-shenshina) [reply etna/models/sarimax.py:185]: seems like I don't understand the problem
could you please give a minimal example to show different behavior?

File: etna/models/sarimax.py [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+            )
+        if horizon:
+            short_regressors = [regressor for regressor in regressor_columns if df[regressor].count() < horizon]
+            if short_regressors:
+                raise ValueError(
+                    f"Regrssors {short_regressors} are too short for chosen horizon value.\n "
+                    "Try lower horizon value, or drop this regressors."
+                )
+
+    def _select_regressors(self, df: pd.DataFrame) -> pd.DataFrame:
```
Comment (julia-shenshina) [inline etna/models/sarimax.py:270]: as soon as `_select_regressors` can return `None` it should be
```suggestion
    def _select_regressors(self, df: pd.DataFrame) -> Optional[pd.DataFrame]:
```

File: etna/models/sarimax.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/models/sarimax.py:185]: Hmm, strange, while testing your variant in the morning I caught the exception, but I can not reproduce it now)

**Review by julia-shenshina — APPROVED** [review]
👍🏼

**Pull request merged**